checkout-operator.mk

```diff
@@ -3,7 +3,9 @@
 #  2 - github repo name (operator name)
 define checkout_operator
 	mkdir -p $(SOURCE_DIR); \
-	pushd $(SOURCE_DIR); \
-	git clone -b master https://github.com/$(1)/$(2).git || (pushd $(2) && git pull && popd); \
-	popd
+	if [[ ! -d $(SOURCE_DIR)/$(2) ]]; then \
+		git clone -b master https://github.com/$(1)/$(2).git $(SOURCE_DIR)/$(2) ;\
+	else \
+		cd $(SOURCE_DIR)/$(2) git reset --hard && git pull --force && cd ../.. ;\
```

Comment:
The reset hard and force pull are actually a HUGE pain if you're doing development.  I had local checkout of the `dedicated-admin-operator` I simply overrode locally in this checkout while I was figuring out what worked with olm.  If this was in there I would need to update the makefile to remove the checkout of source.  Maybe a flag to not have it do this?  Having it as default behavior would be OK.

Reply:
That's a good point. I didn't think about doing development work within this checkout. Flag forthcoming.